core/templates/components/state-directives/outcome-editor/outcome-editor.component.ts

```diff
@@ -134,10 +134,12 @@ export class OutcomeEditorComponent implements OnInit {
   }
 
   isSelfLoopDestStuck(outcome: Outcome): boolean {
+    if (outcome.destIfReallyStuck === null) {
```

Comment:
In what scenario is this null?

Reply:
Intially outcome.destIfReallyStuck is `null`
and stateEditorService.getActiveStateName() gives '`Introduction`'.

but we need   `isSelfLoopDestStuck` to be true for the first time .
for some html tags to be render in its HTML file.

Reply:
Hm, do we know why it is null initially i.e. do we explicitly set it? Do you mean it is null and then it changes value to something else?

Reply:
Hi @kevintab95 
Initially, we define `destIfReallyStuck`  as   
   ` destIfReallyStuck: string | null;`  in  `OutcomeObjectFactory.ts`
   
   >Do you mean it is null and then it changes value to something else?
   Yes
   
   Thanks!

Reply:
Hm sounds like there is a race condition. Could you please explain what is the scenario where we explicitly set it as null? (note that the one you pointed out is just a type definition, not assignment)